bot/seasons/valentines/be_my_valentine.py

```diff
@@ -0,0 +1,235 @@
+                message = f"You cannot send a valentine to {user} as he/she does not have the lovefest role!"
+                return await ctx.send(message)
+
+        if user == ctx.author:
+            # Well a user cant valentine himself/herself.
+            return await ctx.send('Come on dude, you cant send a valentine to yourself :expressionless:')
+
+        guild = self.bot.get_guild(id=Client.guild)
+        emoji_1, emoji_2 = self.random_emoji()
+        lovefest_role = discord.utils.get(guild.roles, id=Lovefest.role_id)
+        valentine, title = self.valentine_check(valentine_type)
+
+        if user is None:
+            author = ctx.author
+            user = self.random_user(author, lovefest_role.members)
+            if user is None:
+                return await ctx.send("There are no users avilable to whome your valentine can be sent.")
+
+        embed = discord.Embed(
+            title=f'{emoji_1}{title} {user.display_name}{emoji_2}',
+            description=f'{valentine} \n **{emoji_2}From anonymous{emoji_1}**',
+            color=Colours.pink
+        )
+        await ctx.author.send(f"Your message has been sent to {user}")
+        await user.send(embed=embed)
+
+    def valentine_check(self, valentine_type):
+        if valentine_type is None:
+            valentine, title = self.random_valentine()
+
+        elif valentine_type.lower() in ['p', 'poem']:
+            valentine = self.valentine_poem()
+            title = 'A poem dedicated to'
+
+        elif valentine_type.lower() in ['c', 'compliment']:
+            valentine = self.valentine_compliment()
+            title = 'A compliment for'
+
+        else:
+            # in this case, the user decides to type his own valentine.
+            valentine = valentine_type
+            title = 'A message for'
+        return valentine, title
+
+    @staticmethod
+    def random_user(author, members):
+        # just making sure that the random does not pick up the same user(ctx.author)
+
+        if members.__len__() != 0:
+            if author in members:
+                members.remove(author)
+            user = random.choice(members)
+        else:
+            user = None
+        return user
+
+    @staticmethod
+    def random_emoji():
+        EMOJI_1 = random.choice(HEART_EMOJIS)
+        EMOJI_2 = random.choice(HEART_EMOJIS)
+        return EMOJI_1, EMOJI_2
+
+    def random_valentine(self):
+        """
+        Grabs a random poem or a compliment (any message).
+        """
+        valentine_poem = random.choice(self.valentines['valentine_poems'])
+        valentine_compliment = random.choice(self.valentines['valentine_compliments'])
+        random_valentine = random.choice([valentine_compliment, valentine_poem])
+        if random_valentine == valentine_poem:
+            title = 'A poem dedicated to'
+        else:
+            title = 'A compliment for '
+        return random_valentine['message'], title
```

Comment:
As the structure of the json data has changed, your `random_valentine` is actually already the message you wanted. Trying to fetch the value of a `message` key here is actually attempting to get it from a string value, and raises an exception, making the command fail. Best thing to do here is just remove that bit:
```suggestion
        return random_valentine, title
```